air/src/virtual_column.rs

```diff
@@ -4,7 +4,7 @@ use core::ops::Mul;
 use p3_field::{AbstractField, Field};
 
 /// An affine function over columns in a PAIR.
-pub struct VirtualPairCol<F: Field> {
+pub struct VirtualPairCol<F: AbstractField> {
```

Comment:
I don't quite get the need for abstract here, wouldn't a virtual column always have concrete weights? Though I guess it doesn't hurt if it's useful.

Does it have to do with `PermutationAirBuilder::EF` being abstract? I think we'll eventually need a concrete extension field in `PermutationAirBuilder`

Reply:
Reverting back, and adding both abstract and concrete extension field types in `PermutationAirBuilder`